test-js/unit/pentaho/visual/base/View.Spec.js

```diff
@@ -34,155 +35,313 @@ define([
-          done();
+        return view.update().then(function() {
+          expect(listeners.didUpdate).not.toHaveBeenCalled();
+        }, function(reason) {
+          expectValidationError(reason.message, "Some error");
+          expect(listeners.rejectedUpdate).toHaveBeenCalled();
         });
+      });
+
+      it("should call the `rejected:update` event listener if `_update` throws", function() {
+        createView(UpdateErrorView);
 
+        return view.update().then(function() {
+          expect(listeners.didUpdate).not.toHaveBeenCalled();
+        }, function(reason) {
+          expectError(reason.message, "Some error");
+          expect(listeners.rejectedUpdate).toHaveBeenCalled();
+        });
       });
 
-      it("`render()` should not invoke `_render` if `_validate` throws", function(done) {
-        var DerivedView = View.extend({
-          _validate: function(){ throw new Error("Some error"); },
-          _render: function(){ return "Rendered"; }
+      it("should invoke `_update` if the view is valid", function() {
+        createView(DerivedView);
+
+        spyOn(view, '_update').and.callThrough();
+
+        return view.update().then(function() {
+          expect(view._update).toHaveBeenCalled();
         });
-        var view = new DerivedView(element, model);
+      });
 
-        spyOn(view, '_render').and.callThrough();
+      it("should create the visualization DOM element if the view is valid", function() {
+        createView(DerivedView);
+        spyOn(view, '_setDomNode').and.callThrough();
 
-        view.render().then(function resolved() {
-          done.fail();
-        }, function rejected(reason) {
-          expect(reason).toBe("Some error");
-          expect(view._render).not.toHaveBeenCalled();
-          done();
+        expect(view.domNode).toBeNull();
+        return view.update().then(function() {
+          expect(view._setDomNode).toHaveBeenCalled();
+          expect(view.domNode instanceof HTMLElement).toBe(true);
         });
 
       });
 
+      it("should emit a 'did:create' event after the first update", function() {
+        createView(DerivedView);
+        var emitted = false;
 
+        listeners.didCreate.and.callFake(function() {
+          emitted = true;
+        });
+
+        return view.update().then(function() {
+          expect(emitted).toBe(true);
+        });
+
+      });
+
+      it("should not invoke `_update` if the view is invalid", function() {
+        createView(ValidationErrorView);
+        spyOn(view, '_update').and.callThrough();
+
+        return view.update().then(function() {
+          expect(listeners.didUpdate).not.toHaveBeenCalled();
+        }, function(reason) {
+          expectValidationError(reason.message, "Some error");
+          expect(view._update).not.toHaveBeenCalled();
+        });
+
+      });
+
+      it("should not create the visualization DOM element if the view is invalid", function() {
```

Comment:
This is not, directly, a requirement :-) Unless you just want to say that _update is not called if the model is not valid.